fileio/easyio.go

```diff
@@ -32,25 +33,63 @@ func EasyOpen(filename string) *EasyReader {
 	if strings.Contains(filename, "http") {
 		return EasyHttp(filename)
 	}
+
 	answer := EasyReader{}
-	answer.File = MustOpen(filename)
-	var err error
+	var hasMagicGzip bool
+	var readerInput io.Reader
+
+	if strings.HasPrefix(filename, "stdin") {
+		// when reading stdin we will assume the input is gzipped
+		// if the file begins with the two magic gzip bytes 1f8d.
+		// If it does, append .gz to the filename so it is parsed
+		// as gzip in the following switch case.
+		answer.File = os.Stdin
+		readerInput, hasMagicGzip = newStdinMagicReader(magicGzip)
+		if hasMagicGzip {
```

Comment:
I think this is the core of the functionality that is added.  If the user tells us to read from "stdin" instead of "stdin.gz" we are now smart enough to realize that they probably wanted "stdin.gz" and seamlessly make this silent change for them.
Your work is technically impressive, but I wonder if a more general implementation would have advantages.  I wonder if right now we are in a bit of a middle ground where we are adding a decent number of lines of code and complexity, to catch a mistake of "stdin" instead of "stdin.gz".  What if we sent inputs into a buffer where we can peek the magic bytes.  We can then either use that buffer if it is plain text, or if we detect magic bytes during the peek we can put it into a gzip buffer and back into a buffered reader we can call ReadString on.
Another option would be to believe the file extension when there is one (one buffer for plain text and two buffers if gzip) and otherwise we would still have one buffer for plain text or three buffers if the file with no extension (e.g. /dev/stdin) was given.  I think this could be added to the easyreader and would not need many more lines and might be a more general solution that could see more uses?  Are there performance differences?